UnityProject/Assets/Scripts/Clothing/Eyewear/NightVisionGoggles.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using CameraEffects;
+
+public class NightVisionGoggles : MonoBehaviour, IServerInventoryMove
+{
+	[SerializeField, Tooltip("How far the player will be able to see in the dark while he has the goggles on.")]
+	private Vector3 nightVisionVisibility;
+
+	[SerializeField, Tooltip("The default minimal visibility size.")]
+	private Vector3 defaultVisionVisibility;
+
+
+	public void OnInventoryMoveServer(InventoryMove info)
+		{
+			RegisterPlayer registerPlayer;
+
+			if (info.ToSlot != null && info.ToSlot?.NamedSlot != null)
+			{
+				registerPlayer = info.ToRootPlayer;
+
+				if (registerPlayer != null && info.ToSlot.NamedSlot == NamedSlot.eyes)
+				{
+					OnWearing();
+				}
+			}
+
+			if (info.FromSlot != null && info.FromSlot?.NamedSlot != null && info.ToSlot != null)
+			{
+				registerPlayer = info.FromRootPlayer;
+
+				if (registerPlayer != null && info.FromSlot.NamedSlot == NamedSlot.eyes)
+				{
+					OnTakingOff();
+				}
+			}
+		}
+	
+	private void OnTakingOff()
+		{
+			var camera = Camera.main;
+			camera.GetComponent<CameraEffectControlScript>().ToggleNightVisionEffectState();
+			camera.GetComponent<CameraEffectControlScript>().AdjustPlayerVisibility(defaultVisionVisibility);
+		}
+
+	private void OnWearing()
+		{
+			var camera = Camera.main;
+			camera.GetComponent<CameraEffectControlScript>().ToggleNightVisionEffectState();
+			camera.GetComponent<CameraEffectControlScript>().AdjustPlayerVisibility(nightVisionVisibility);
+		}
+}
```

Comment:
This is serverside only, and wont affect clients. You'll need to make a TargetRPC

Reply:
just put `[TargetRpc]` above the function, right?

Reply:
Add it above OnTakingOff and OnWearing, you'll need to rename them to TargetOnTakingOff  and TargetOnWearing

Reply:
They'll also need this ```NetworkConnection target``` parameter added

Reply:
do i need to change anything inside the function as well?

Reply:
nope